src/file/SDL_iostream.c

```diff
@@ -360,7 +361,10 @@ static Sint64 SDLCALL stdio_seek(void *userdata, Sint64 offset, SDL_IOWhence whe
     }
 #endif
 
-    if (fseek(iodata->fp, (fseek_off_t)offset, stdiowhence) == 0) {
+    /* don't make a possibly-costly API call for the noop seek from SDL_TellIO */
+    is_noop = (whence == SDL_IO_SEEK_CUR && offset == 0);
```

Comment:
In SDL3 (but not SDL2!) we allow in-place variable declarations, so this can be:

```c
const SDL_bool is_noop = (whence == SDL_IO_SEEK_CUR) && (offset == 0);
```

(Note that I changed the parentheses too, just to be clearer about how this logic separates.)

Reply:
Good to know regarding declarations. It'll be fun for me to keep this straight in the future. Fixed the parens here also. Thanks for the reviews!